src/main/webapp/js/client.js

```diff
@@ -155,3 +155,111 @@ function getRestaurants() {
     // Using Promise here to mimick API response from the backend
     return Promise.resolve(restaurants);
 }
+
+function getSignIn() {
+  fetch('/login').then(response => response.json()).then((loginInfo) => {
+    if (loginInfo.length == 1) { // means user is not logged in
+      //var parent = document.getElementById("para");
+      var loginLink = loginInfo[0];
+      var message = "Go " + loginLink + " to sign in";
+      var para = document.createElement("p");
+      para.appendChild(document.createTextNode(message));
+      document.body.appendChild(para);
+      window.location.href = loginLink;
+    } else if (loginInfo.length ==2) {
+      // user has yet to register
+      // direct user to a new page asking them to choose between charity and restaurant
+      var registerLink = "register.html";
+      window.location.href = registerLink;
+    } else {
+      var userEmail = loginInfo[0];
+      var userType = loginInfo[2];
+      var para = document.createElement("p");
+      var message = "You are already logged in as " + userEmail + " of type " + userType;
+      para.appendChild(document.createTextNode(message));
+      document.body.appendChild(para);
+      // create the link to return to main page
+      mainPageUrl = "index.html";
+      var a = document.createElement('a');
+      a.href = mainPageUrl;
+      a.title = mainPageUrl;
+      a.appendChild(document.createTextNode("here"));
+      var para2 = document.createElement("p");
+      para2.appendChild(document.createTextNode("Click "));
+      para2.appendChild(a);
+      para2.appendChild(document.createTextNode("to return to main page"));
+      document.body.appendChild(para2);
+      var logoutLink = loginInfo[1];
+      var a2 = document.createElement('a');
+      a2.href = logoutLink;
+      a2.title = logoutLink;
+      a2.appendChild(document.createTextNode("here"));
+      var para3 = document.createElement("p");
+      para3.appendChild(document.createTextNode("Logout "));
+      para3.appendChild(a2);
+      document.body.appendChild(para3);
+    }
+  })
+} 
+
+function getRegister() {
+  console.log("at register page");
+  fetch('/login').then(response => response.json()).then((loginInfo) => {
```

Comment:
Let's break this method as well

Reply:
mm this method is quite short already though